.proposals/records/index-scaling-v1.md

```diff
@@ -0,0 +1,235 @@
+# Index Scaling: Decoupling disk, reads/writes 
```

Comment:
I sounds nice! The only thing that worries me is the latency between write to gcs and read sync and search hitting a IndexReadReplica not available during this time.

When a write occurs on the system, all read replicas will be stale during: gcs write sync time + reader sync time. In this scenario, a search will potentially call a read replica and we'll need a second round trip to the index (at best) to obtain search results. The current API allow a `shards` parameter with a list of shard replicas to query and the frontend caches the searched shard. If the frontend caches a shard replica of a IndexReadReplica, two round trips are always ensured while not in sync. The number of potential hits to a stale read replica scales with the addition of new replicas.

However, within the assumption we'll have way more reads than writes, this shouldn't be too common.